tests/integration/turtlebot/test_successful_mission.py

```diff
@@ -0,0 +1,150 @@
+import shutil
+import time
+from copy import deepcopy
+from datetime import datetime, timedelta
+from pathlib import Path
+from typing import List
+
+import numpy as np
+import pytest
+from fastapi import FastAPI
+from injector import Injector
+from starlette.testclient import TestClient
+
+from isar.apis.api import API
+from isar.config import config
+from isar.models.mission import Mission
+from isar.modules import (
+    APIModule,
+    CoordinateModule,
+    LocalPlannerModule,
+    LocalStorageModule,
+    QueuesModule,
+    ReaderModule,
+    RequestHandlerModule,
+    RobotModule,
+    ServiceModule,
+    StateMachineModule,
+    UtilitiesModule,
+)
+from isar.services.readers.base_reader import BaseReader
+from robot_interface.models.geometry.frame import Frame
+from robot_interface.models.geometry.position import Position
+from robot_interface.models.mission import DriveToPose
+from tests.isar.state_machine.test_state_machine import StateMachineThread
+from tests.test_modules import MockNoAuthenticationModule
+
+
+@pytest.fixture()
+def injector_turtlebot():
+    config.set("DEFAULT", "robot_package", "isar_turtlebot")
+    config.set("DEFAULT", "local_storage_path", "./tests/results")
+    config.set(
+        "DEFAULT",
+        "predefined_missions_folder",
+        "./tests/integration/turtlebot/config/missions",
+    )
+
+    config.set("DEFAULT", "maps_folder", "tests/integration/turtlebot/config/maps")
+    config.set("DEFAULT", "default_map", "turtleworld")
+
+    return Injector(
+        [
+            APIModule,
+            MockNoAuthenticationModule,
+            CoordinateModule,
+            QueuesModule,
+            ReaderModule,
+            RequestHandlerModule,
+            RobotModule,
+            ServiceModule,
+            StateMachineModule,
+            LocalPlannerModule,
+            LocalStorageModule,
+            UtilitiesModule,
+        ]
+    )
+
+
+@pytest.fixture()
+def state_machine_thread(injector_turtlebot) -> StateMachineThread:
+    return StateMachineThread(injector=injector_turtlebot)
+
+
+@pytest.fixture(autouse=True)
+def run_before_and_after_tests():
+    results_folder: Path = Path("tests/results")
+    yield
+
+    print("Removing temporary results folder for testing")
+    shutil.rmtree(results_folder)
```

Comment:
This gives me a `FileNotFoundError` when 'tests/results' folder does not exist. I had not the simulator running or the isar-turtlebot installed yet though. But still, maybe it should handle the case where the folder does not exist?

Reply:
Done.